native/src/main/java/io/ballerina/stdlib/task/HealthCheckScheduler.java

```diff
@@ -0,0 +1,48 @@
+/*
+ *  Copyright (c) 2025 WSO2 LLC. (http://www.wso2.com).
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+
+package io.ballerina.stdlib.task;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Logger;
+
+public class HealthCheckScheduler {
+
+    private static final ScheduledExecutorService scheduler = Executors.newSingleThreadScheduledExecutor();
+
+    public static void startHealthCheckUpdater(Connection connection, String tokenId, int periodInSeconds) {
+        Runnable task = () -> {
+            String healthCheckQuery = "INSERT INTO health_check(token_id, last_heartbeat) " +
+                    "VALUES (?, CURRENT_TIMESTAMP) ON DUPLICATE KEY UPDATE last_heartbeat = CURRENT_TIMESTAMP";
+            try (PreparedStatement stmt = connection.prepareStatement(healthCheckQuery)) {
+                stmt.setString(1, tokenId);
+                stmt.executeUpdate();
+                connection.commit();
+            } catch (SQLException e) {
+                Logger.getLogger(HealthCheckScheduler.class.getName())
+                        .severe("Error updating health check: " + e.getMessage());
```

Comment:
Logs the full stack trace, not just the message. Better to include contextual information like token id. That will helps to identify the exact issue easily.